### ISSUE TITLE
[ubuntu-22.04/kernel-5.19/RX7900XTX/rocm-5.5.0]  libamdhip.so got segmentfault 

### ISSUE THREAD

**Issue by natugrey** [RadeonOpenCompute/ROCm#2122]
**my env**:
**host**: ubuntu 22.04 with kernel 5.19,   dkms installed
**docker**: rocm/pytorch:latest   ubuntu 22.04 kernel 5.19,  rocm 5.5.0 installed

**run rocminfo inside container:**

  Name:                    gfx1100                            
  Uuid:                    GPU-c7b68b8200000000               
  Marketing Name:          Radeon RX 7900 XTX                 
  Vendor Name:             AMD                                
  Feature:                 KERNEL_DISPATCH                    
  Profile:                 BASE_PROFILE                       
  Float Round Mode:        NEAR                               
  Max Queue Number:        128(0x80)                          
  Queue Min Size:          64(0x40)                           
  Queue Max Size:          131072(0x20000)                    
  Queue Type:              MULTI                              
  Node:                    1                                  
  Device Type:             GPU                                
  Cache Info:              
    L1:                      32(0x20) KB                        
    L2:                      6144(0x1800) KB                    
    L3:                      98304(0x18000) KB                  
  Chip ID:                 29772(0x744c)                      
  ASIC Revision:           0(0x0)                             
  Cacheline Size:          64(0x40)                           
  Max Clock Freq. (MHz):   2526                               
  BDFID:                   2304                               
  Internal Node ID:        1                                  
  Compute Unit:            96                                 
  SIMDs per CU:            2                                  
  Shader Engines:          12                                 
  Shader Arrs. per Eng.:   2                                  
  WatchPts on Addr. Ranges:4                                  
  Features:                KERNEL_DISPATCH 
  Fast F16 Operation:      TRUE                               
  Wavefront Size:          32(0x20)                           
  Workgroup Max Size:      1024(0x400)                        
  Workgroup Max Size per Dimension:
    x                        1024(0x400)                        
    y                        1024(0x400)                        
    z                        1024(0x400)                        
  Max Waves Per CU:        32(0x20)                           
  Max Work-item Per CU:    1024(0x400)                        
  Grid Max Size:           4294967295(0xffffffff)             
  Grid Max Size per Dimension:
    x                        4294967295(0xffffffff)             
    y                        4294967295(0xffffffff)             
    z                        4294967295(0xffffffff)             
  Max fbarriers/Workgrp:   32                                 
  Pool Info:               
    Pool 1                   
      Segment:                 GLOBAL; FLAGS: COARSE GRAINED      
      Size:                    25149440(0x17fc000) KB             
      Allocatable:             TRUE                               
      Alloc Granule:           4KB                                
      Alloc Alignment:         4KB                                
      Accessible by all:       FALSE                              
    Pool 2                   
      Segment:                 GROUP                              
      Size:                    64(0x40) KB                        
      Allocatable:             FALSE                              
      Alloc Granule:           0KB                                
      Alloc Alignment:         0KB                                
      Accessible by all:       FALSE                              
  ISA Info:                
    ISA 1                    
      Name:                    amdgcn-amd-amdhsa--gfx1100         
      Machine Models:          HSA_MACHINE_MODEL_LARGE            
      Profiles:                HSA_PROFILE_BASE                   
      Default Rounding Mode:   NEAR                               
      Default Rounding Mode:   NEAR                               
      Fast f16:                TRUE                               
      Workgroup Max Size:      1024(0x400)                        
      Workgroup Max Size per Dimension:
        x                        1024(0x400)                        
        y                        1024(0x400)                        
        z                        1024(0x400)                        
      Grid Max Size:           4294967295(0xffffffff)             
      Grid Max Size per Dimension:
        x                        4294967295(0xffffffff)             
        y                        4294967295(0xffffffff)             
        z                        4294967295(0xffffffff)             
      FBarrier Max Size:       32                                 

**i was trying to run a torch minist example, and got segmentfault.
code like:**
`
def train(args, model, device, train_loader, optimizer, epoch):
    model.train()
    for batch_idx, (data, target) in enumerate(train_loader):
        data, target = data.to(device), target.to(device)
        optimizer.zero_grad()
        output = model(data)
        loss = F.nll_loss(output, target)
        loss.backward()
        optimizer.step()
        if batch_idx % args.log_interval == 0:
            print('Train Epoch: {} [{}/{} ({:.0f}%)]\tLoss: {:.6f}'.format(
                epoch, batch_idx * len(data), len(train_loader.dataset),
                100. * batch_idx / len(train_loader), loss.item()))
            if args.dry_run:
                break

if __name__ == '__main__':
    '''
    SOME CODE
    ''' 
    use_cuda = not args.no_cuda and torch.cuda.is_available()
    use_mps = not args.no_mps and torch.backends.mps.is_available()

    torch.manual_seed(args.seed)

    if use_cuda:
        device = torch.device("cuda")
    elif use_mps:
        device = torch.device("mps")
    else:
        device = torch.device("cpu")

   \# if uncomment next line ,runs well 
   \# device = torch.device("cpu")

    train_kwargs = {'batch_size': args.batch_size}
    test_kwargs = {'batch_size': args.test_batch_size}
    if use_cuda:
        cuda_kwargs = {'num_workers': 1,
                       'pin_memory': True,
                       'shuffle': True}
        train_kwargs.update(cuda_kwargs)
        test_kwargs.update(cuda_kwargs)

    transform=transforms.Compose([
        transforms.ToTensor(),
        transforms.Normalize((0.1307,), (0.3081,))
        ])
    dataset1 = datasets.MNIST('../data', train=True, download=True,
                       transform=transform)
    dataset2 = datasets.MNIST('../data', train=False,
                       transform=transform)
    train_loader = torch.utils.data.DataLoader(dataset1,**train_kwargs)
    test_loader = torch.utils.data.DataLoader(dataset2, **test_kwargs)

    model = Net().to(device)
    optimizer = optim.SGD(model.parameters(), lr=args.lr)

    scheduler = StepLR(optimizer, step_size=1, gamma=args.gamma)
    for epoch in range(1, args.epochs + 1):
        train(args, model, device, train_loader, optimizer, epoch)
        test(model, device, test_loader)
        scheduler.step()
`

**The code  stop at 'optimizer.zero_grad()', and if i use cpu, code runs well.**

**And I run 'dmesg' on host, I got**
`
[37051.819197] python[18023]: segfault at 20 ip 00007fd538015415 sp 00007ffd5a736720 error 4 in libamdhip64.so.5.5.50500[7fd537f4d000+388000]
[37051.819204] Code: 85 db 0f 88 1d 01 00 00 48 8b 45 98 48 8b 50 68 48 8b 40 70 48 89 45 90 48 29 d0 48 c1 f8 03 48 39 c3 73 31 4c 8b 24 da 31 c0 <41> 80 7c 24 20 00 74 56 48 8b 4d c8 64 48 33 0c 25 28 00 00 00 0f
[37204.360121] python[18193]: segfault at 20 ip 00007f1c171c5415 sp 00007ffcc96a2650 error 4 in libamdhip64.so.5.5.50500[7f1c170fd000+388000]
[37204.360128] Code: 85 db 0f 88 1d 01 00 00 48 8b 45 98 48 8b 50 68 48 8b 40 70 48 89 45 90 48 29 d0 48 c1 f8 03 48 39 c3 73 31 4c 8b 24 da 31 c0 <41> 80 7c 24 20 00 74 56 48 8b 4d c8 64 48 33 0c 25 28 00 00 00 0f
[38115.251414] python[20086]: segfault at 20 ip 00007f4227c6f415 sp 00007fffbc6dc9c0 error 4 in libamdhip64.so.5.5.50500[7f4227ba7000+388000]
[38115.251423] Code: 85 db 0f 88 1d 01 00 00 48 8b 45 98 48 8b 50 68 48 8b 40 70 48 89 45 90 48 29 d0 48 c1 f8 03 48 39 c3 73 31 4c 8b 24 da 31 c0 <41> 80 7c 24 20 00 74 56 48 8b 4d c8 64 48 33 0c 25 28 00 00 00 0f
`

**Wish to get some help, thank you.**

**Comment by JanzenJohn** [comment]
how did you fix it?

**Comment by natugrey** [comment]
> 
solved [https://github.com/bigchirv/docker_sd_webui_gfx1100](here).
Need to compile pytorch yourself.
